zebra/zapi_msg.c

```diff
@@ -1586,8 +1586,45 @@ static void zread_route_add(ZAPI_HANDLER_ARGS)
 	if (CHECK_FLAG(api.message, ZAPI_MESSAGE_SRCPFX))
 		src_p = &api.src_prefix;
 
+
+	/**
+	 * zebra_nhg cached handling to speed up route installs
+	 * for routes being sent with the same nexthops in a row.
+	 */
+	assert(sizeof(api.nexthops) == sizeof(client->inhg_cache.nexthops));
+
+	/* If matches cached nexthops, use that NHE ID now */
+	if (memcmp(api.nexthops, client->inhg_cache.nexthops,
+		   sizeof(client->inhg_cache.nexthops))
```

Comment:
could we just compare the exact number of nexthops that have actually been decoded into `api` ? if we captured the number of nexthops in the client's cache struct, that could be compared before the (more expensive) memcmp.

Reply:
good idea, will do